internals/overlord/cmdstate/request.go

```diff
@@ -117,12 +117,15 @@ func Exec(st *state.State, args *ExecArgs) (*state.Task, ExecMetadata, error) {
 		environment["LANG"] = "C.UTF-8"
 	}
 
-	// Set default working directory to $HOME, or / if $HOME not set.
+	// Set default working directory to $HOME, or / if $HOME not set or doesn't exist.
 	workingDir := args.WorkingDir
 	if workingDir == "" {
 		workingDir = environment["HOME"]
 		if workingDir == "" {
 			workingDir = "/"
+		} else if !osutil.IsDir(workingDir) {
```

Comment:
maybe we should do an `os.Stat` check first. It might be useful to distinguish between "not found" and "found but it's a file, not a dir", and surface a slightly different error message to the user.

Reply:
Done.